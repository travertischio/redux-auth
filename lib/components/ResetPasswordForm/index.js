'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactIntl = require('react-intl');

var _immutable = require('redux-form/immutable');

var _reduxForm = require('react-form-fields/lib/PasswordField/reduxForm');

var _reduxForm2 = _interopRequireDefault(_reduxForm);

var _required = require('validators/lib/required');

var _setOfPasswordValidators = require('validators/lib/setOfPasswordValidators');

var _matchToNewPassword = require('validators/lib/matchToNewPassword');

var _messages = require('./messages');

var _messages2 = _interopRequireDefault(_messages);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function ResetPasswordForm(props) {
  var handleSubmit = props.handleSubmit,
      pristine = props.pristine,
      submitting = props.submitting;
  var formatMessage = props.intl.formatMessage;

  var newPasswordLabel = formatMessage(_messages2.default.newPasswordLabel);
  var newPasswordPlaceholder = formatMessage(_messages2.default.newPasswordPlaceholder);
  var reNewPassword = formatMessage(_messages2.default.reNewPassword);
  var reNewPasswordPlaceholder = formatMessage(_messages2.default.reNewPasswordPlaceholder);

  var onSubmit = function onSubmit(event) {
    if (event && event.preventDefault) {
      event.preventDefault();
    }

    handleSubmit();
  };

  return _react2.default.createElement(
    'form',
    { onSubmit: onSubmit, noValidate: true },
    _react2.default.createElement(_immutable.Field, {
      id: 'newPassword',
      name: 'newPassword',
      label: newPasswordLabel,
      placeholder: newPasswordPlaceholder,
      validate: _setOfPasswordValidators.setOfPasswordValidators,
      component: _reduxForm2.default
    }),
    _react2.default.createElement(_immutable.Field, {
      id: 'reNewPassword',
      name: 'reNewPassword',
      label: reNewPassword,
      placeholder: reNewPasswordPlaceholder,
      validate: [_required.required, _matchToNewPassword.matchToNewPassword],
      component: _reduxForm2.default
    }),
    _react2.default.createElement(
      'div',
      null,
      _react2.default.createElement(
        'button',
        {
          type: 'submit',
          disabled: pristine || submitting
        },
        'Save'
      )
    )
  );
} /**
  *
  * ResetPasswordForm
  *
  */

ResetPasswordForm.propTypes = {
  handleSubmit: _propTypes2.default.func.isRequired,
  intl: _propTypes2.default.object.isRequired,
  pristine: _propTypes2.default.bool.isRequired,
  submitting: _propTypes2.default.bool.isRequired
};

exports.default = (0, _immutable.reduxForm)({
  form: 'resetPasswordForm'
})((0, _reactIntl.injectIntl)(ResetPasswordForm));