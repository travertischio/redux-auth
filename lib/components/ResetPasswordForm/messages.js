"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _reactIntl = require("react-intl");

/*
 * ResetPasswordForm Messages
 *
 * This contains all the text for the ResetPasswordForm component.
 */
var _default = (0, _reactIntl.defineMessages)({
  newPasswordLabel: {
    id: 'app.components.SignInForm.newPasswordLabel',
    defaultMessage: 'New password'
  },
  newPasswordPlaceholder: {
    id: 'app.components.SignInForm.newPasswordPlaceholder',
    defaultMessage: 'Enter new password'
  },
  reNewPassword: {
    id: 'app.components.SignInForm.reNewPassword',
    defaultMessage: 'Retype new password'
  },
  reNewPasswordPlaceholder: {
    id: 'app.components.SignInForm.Placeholder',
    defaultMessage: 'Retype new password'
  }
});

exports["default"] = _default;