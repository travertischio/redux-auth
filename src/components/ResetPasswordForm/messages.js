/*
 * ResetPasswordForm Messages
 *
 * This contains all the text for the ResetPasswordForm component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  newPasswordLabel: {
    id: 'app.components.SignInForm.newPasswordLabel',
    defaultMessage: 'New password',
  },
  newPasswordPlaceholder: {
    id: 'app.components.SignInForm.newPasswordPlaceholder',
    defaultMessage: 'Enter new password',
  },
  reNewPassword: {
    id: 'app.components.SignInForm.reNewPassword',
    defaultMessage: 'Retype new password',
  },
  reNewPasswordPlaceholder: {
    id: 'app.components.SignInForm.Placeholder',
    defaultMessage: 'Retype new password',
  },
});
