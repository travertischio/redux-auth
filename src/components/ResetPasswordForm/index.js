/**
*
* ResetPasswordForm
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import {
  Field,
  reduxForm,
} from 'redux-form/immutable';
import PasswordField from 'react-form-fields/lib/PasswordField/reduxForm';
import { required as requiredValidator } from 'validators/lib/required';
import { setOfPasswordValidators } from 'validators/lib/setOfPasswordValidators';
import { matchToNewPassword as matchToNewPasswordValidator } from 'validators/lib/matchToNewPassword';
import messages from './messages';

function ResetPasswordForm(props) {
  const {
    handleSubmit,
    pristine,
    submitting,
    valid,
  } = props;
  const { formatMessage } = props.intl;
  const newPasswordLabel = formatMessage(messages.newPasswordLabel);
  const newPasswordPlaceholder = formatMessage(messages.newPasswordPlaceholder);
  const reNewPassword = formatMessage(messages.reNewPassword);
  const reNewPasswordPlaceholder = formatMessage(messages.reNewPasswordPlaceholder);

  const onSubmit = (event) => {
    if (event && event.preventDefault) {
      event.preventDefault();
    }

    if (valid) {
      handleSubmit();
    }
  };

  return (
    <form onSubmit={onSubmit} noValidate>
      <Field
        id="newPassword"
        name="newPassword"
        label={newPasswordLabel}
        placeholder={newPasswordPlaceholder}
        validate={setOfPasswordValidators}
        component={PasswordField}
      />
      <Field
        id="reNewPassword"
        name="reNewPassword"
        label={reNewPassword}
        placeholder={reNewPasswordPlaceholder}
        validate={[requiredValidator, matchToNewPasswordValidator]}
        component={PasswordField}
      />
      <div>
        <button
          type="submit"
          disabled={pristine || submitting}
        >
          Save
        </button>
      </div>
    </form>
  );
}

ResetPasswordForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  pristine: PropTypes.bool.isRequired,
  submitting: PropTypes.bool.isRequired,
  valid: PropTypes.bool.isRequired,
  intl: PropTypes.object.isRequired,
};

export default reduxForm({
  form: 'resetPasswordForm',
})(injectIntl(ResetPasswordForm));
