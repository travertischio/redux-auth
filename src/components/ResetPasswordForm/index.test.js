import React from 'react';
import { createComponentWithIntl } from 'react-unit-testing-utils';

import ResetPasswordForm from '.';

describe('<ResetPasswordForm />', () => {
  it('should render ResetPasswordForm', () => {
    const initialState = {};
    const { component } = createComponentWithIntl(<ResetPasswordForm />, initialState);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
