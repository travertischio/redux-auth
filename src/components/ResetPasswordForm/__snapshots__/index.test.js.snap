// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResetPasswordForm /> should render ResetPasswordForm 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div>
    <label
      htmlFor="new_password"
    >
      New password
    </label>
    <div
      className=""
    >
      <input
        id="new_password"
        name="new_password"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Enter new password"
        type="password"
        value=""
      />
    </div>
  </div>
  <div>
    <label
      htmlFor="re_new_password"
    >
      Retype new password
    </label>
    <div
      className=""
    >
      <input
        id="re_new_password"
        name="re_new_password"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Retype new password"
        type="password"
        value=""
      />
    </div>
  </div>
  <div>
    <button
      disabled={true}
      type="submit"
    >
      Save
    </button>
  </div>
</form>
`;
